### PR TITLE
reformatting task coding challenge

### PR DESCRIPTION
**Pull request by waheeduddin**
Please consider this my PR for the task 'vairable_data' in the 'gradients' submodule.
The task number #9283 and the ToDo list is #776

close https://github.com/unifyai/ivy/issues/9283

### CODE REVIEW

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hi
Thanks for contributing to ivy😊
An overview of this task is that these functions won't need to have array and container methods so you can remove them.
And as the functions are private, this means they won't be rendered in docs we can write simple docstring explaining functions with no need for putting examples.

Thanks🙂

File: ivy/functional/ivy/gradients.py [hunk 1]
```diff
@@ -268,8 +269,79 @@ def _is_variable(x, exclusive=False) -> bool:
     )
 
 
-def _variable_data(x):
+@outputs_to_ivy_arrays
+def _variable_data(x: Union[ivy.Array, ivy.NativeArray]) -> ivy.Array:
```
[the remark this answers is not in the document]
Comment (waheeduddin) [reply ivy/functional/ivy/gradients.py:273]: Thank you so much for pointing this out. I had the same ideas but i went all the way to complete the general requirements of 'reformatting task' as explained in the deep dive.

I will remove the examples and remove the array and container implementations. Will be looking forward to your review again. Thanks a lot!

[the remark this answers is not in the document]
Comment (waheeduddin) [reply ivy/functional/ivy/gradients.py:273]: I have made the requested changes. Please review once more. 
Thank you very much for the help and support.

File: ivy/functional/ivy/gradients.py [hunk 2]
```diff
@@ -268,7 +269,21 @@ def _is_variable(x, exclusive=False) -> bool:
     )
 
 
-def _variable_data(x):
+@outputs_to_ivy_arrays
```
Comment (hirwa-nshuti) [inline ivy/functional/ivy/gradients.py:272]: This decorator is not needed as this function is to be used inside other functions

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey
All looks good so far just a few changes to be made😊

Thanks

File: ivy/functional/ivy/gradients.py (hunk 2 above)
Comment (waheeduddin) [reply ivy/functional/ivy/gradients.py]: I have removed the decorator in the latest changes. Thank you!

**Comment by waheeduddin** [comment]
Thank you for your patience. I have removed the decorator. Please review the work at your convenience.
Warm regards.

File: ivy/functional/ivy/gradients.py [hunk 3]
```diff
@@ -268,11 +268,27 @@ def _is_variable(x, exclusive=False) -> bool:
...
+    -------
+    ret
+        An array with contents of the input.
+    """
     x = ivy.to_native(x, nested=True)
-    return ivy.nested_map(
+    ret = ivy.nested_map(
         x, lambda x: current_backend(x).variable_data(x), include_derived=True
     )
+    return ivy.nested_map(ret, ivy.to_ivy, include_derived=True)
```
Comment (hirwa-nshuti) [inline ivy/functional/ivy/gradients.py:291]: Why do you think that this extra line should be added?

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey
Check below comment🙂

File: ivy/functional/ivy/gradients.py (hunk 3 above)
Comment (waheeduddin) [reply ivy/functional/ivy/gradients.py:291]: Yes. So all the backend 'variable_data' methods return instances native to each individual backend framework. In order to make the output of the 'Ivy.gradients._variable_data' function compatible with other Ivy functions, I added this line (line no. 291) to make sure the function returns Ivy instances and not backend specific instances.

Comment (waheeduddin) [reply ivy/functional/ivy/gradients.py:291]: Is it the right approach @hirwa-nshuti ?

Comment (hirwa-nshuti) [reply ivy/functional/ivy/gradients.py:291]: Yeah makes sense🙂

**Review by hirwa-nshuti — APPROVED** [review]
LGTM!
Thanks for contributing😊

**Pull request merged**